src/fp.cpp

```diff
@@ -210,10 +279,10 @@ fp fp::inverse() const
 
     if(r.cmp(MODULUS) >= 0 || z > 0)
     {
-        _lsubAssign(&r, &MODULUS);
+        _lsubtract(&r, &r, &MODULUS);
     }
     u = MODULUS;
-    _lsubAssign(&u, &r);
+    _lsubtract(&u, &u, &r);
```

Comment:
At this point, I believe it is possible for `u` to be greater than or equal to the prime modulus. For example, consider the case where `r` was zero at this point (not sure if it is possible to reach this point with a `r` of zero). Or consider the case where `r` is greater than the prime modulus (again not sure if that is possible).

If this is true, then this is concerning given that it will be used in a call to `_double` below. However, it may be safe to call `_double` with a field element greater than or equal to the prime modulus and still get a mathematically sensible output. But has that been verified to be the case?

Reply:
Checked and optimized. See the comments in the function.